mcs/class/System.Drawing/System.Drawing/Color.cs

```diff
@@ -151,7 +151,21 @@ public static Color FromArgb (int argb)
 
 		public static Color FromKnownColor (KnownColor color)
 		{
-			return KnownColors.FromKnownColor (color);
+			Color c;
+			short n = (short)kc;
+			if ((n <= 0) || (n >= ArgbValues.Length)) {
+				// This is what it returns!
+				c = Color.FromArgb (0, 0, 0, 0);
+				c.state |= (short) Color.ColorType.Named;
+			} else {
+				c = new Color ();
+				c.state = (short) (Color.ColorType.ARGB | Color.ColorType.Known | Color.ColorType.Named);
+				if ((n < 27) || (n > 169))
+					c.state |= (short) Color.ColorType.System;
+				c.Value = ArgbValues [n];
+			}
+			c.knownColor = n;
+			return c;
 		}
 
 		public static Color FromName (string name)
```

Comment:
This one uses KnownColors.FromKnownColor which could use FromKnownColor now directly.